main.py

```diff
@@ -258,6 +264,10 @@ def updates_join():
         flask.flash("Please enter your email to subscribe to updates", 'danger')
         return flask.redirect("/candidates")
 
+    if 'postcode' not in flask.session:
```

Comment:
Arguably they could sign up for updates without postcode, but probably better to force it.

Maybe a success flash saying something like "Enter your postcode before signing up for updates"